php/tests/VerifierTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+use Ramsey\Uuid\Uuid;
+use TrueLayer\Signing\Signer;
+use TrueLayer\Signing\Verifier;
+use TrueLayer\Signing\Tests\MockData;
+
+it('should validate a valid signature', function () {
+    $keys = MockData::generateKeyPair();
+    $signer = Signer::signWithKey(Uuid::uuid4(), $keys['private'], null);
+    $verifier = Verifier::verifyWithKey($keys['public']);
+
+    $signature = $signer->method("PUT")
+        ->path("/test")
+        ->header("X-Idempotency-Key", "idempotency-test")
+        ->body('{"random-key": "random-value"}')
+        ->sign();
+
+    $verifier->method('PUT')
+        ->path('/test')
+        ->header('X-Idempotency-Key', 'idempotency-test')
+        ->body('{"random-key": "random-value"}')
+        ->requireHeaders([
+            'X-Idempotency-Key'
+        ]);
+
+    expect($verifier->verify($signature))->not->toThrow(Exception::class);
+});
+
```

Comment:
One important scenario is the static signature verify, see https://github.com/TrueLayer/truelayer-signing/blob/0d252d4906f77bce2aaa9352c30df20d729ad52f/rust/tests/usage.rs#L87..L102

If every lang ensure that this exact signature works for the verify impl then we ensure that all langs have consistent signing implementations. Otherwise it can be quite easy to make a signing impl that works, but isn't exactly the same as the other langs, and therefore won't actually work when it hits our rust/c# servers.

Reply:
Done!